### DISCUSSION TITLE
Restrict embed to specific domain?

### DISCUSSION THREAD

**Discussion by RevolutionHarbor** [mediacms-io/mediacms#875] [Feature Requests]
How can I add a custom state? 
I'd like to have a private video that is only allowed to be embedded on specific domains. You can see an example of this in Vimeo.

 - Public
 - Unlisted
 - Private
- custom* allow embed on [customdomain.com]

**Comment by mgogoulos** [comment]
You can add a new state on files.models.MEDIA_STATES , and then you have to run makemigrations && migrate Django commands to get the new values populated. Then adjust the logic on files.methods.get_next_state on how `save` action works (this is to ensure for example that a user that is not editor or admin cannot make a media public, if the media state is unlisted for example, and this is the default workflow for the portal - while for example they can make their media from published to unlisted or private)

This will allow the new state to appear on the list while editing a Media. Since this is a new state, you will have to see how the field `listable` is used in files.models and probably make adjustements there. `listable` is used in querysets to consider whether to include or not medias. So in a case that used to be


```
if self.state == "public
...
```

you now might have to do a 

```
if self.state in ["public", "new_custom_state_here"]
```

And then of course it would make sense to adjust the code that prevents a media from getting embedded by another domain (most probably done through nginx settings?)

There's room for refactoring here to bring these pieces on a single place, so feel free to consider opening a PR!!!

Regards
